docs/learn/bitesize-invaders.md

```diff
@@ -0,0 +1,309 @@
+//and when their lobby scenes are finished loading.
+if (IsServer)
+{
+    m_AllPlayersInLobby = false;
+
+    //Server will be notified when a client connects
+    NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnectedCallback;
+    SceneTransitionHandler.sceneTransitionHandler.OnClientLoadedScene += ClientLoadedScene;
+}
+```
+
+<!-- ```csharp reference
+https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/LobbyControl.cs#L40-L49
+``` -->
+
+Whenever the `OnClientLoadedScene` callback is called, the custom `ClientLoadedScene` function is also called. And that is the location where you add the new Player to a container that just loaded the Lobby Scene, generate user stats for it (which is just a random name), and then later send an update to the rest of the users notifying them that someone new has joined the Lobby. (line 102119
+
+```csharp
+/// <summary>
+///     ClientLoadedScene
+///     Invoked when a client has loaded this scene
+/// </summary>
+/// <param name="clientId"></param>
+private void ClientLoadedScene(ulong clientId)
+{
+    if (IsServer)
+    {
+        if (!m_ClientsInLobby.ContainsKey(clientId))
+        {
+            m_ClientsInLobby.Add(clientId, false);
+            GenerateUserStatsForLobby();
+        }
+
+        UpdateAndCheckPlayersInLobby();
+    }
+}
+```
+
+<!-- ```csharp reference
+https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/LobbyControl.cs#L102-L119
+``` -->
+
+When the players join this Lobby, they all need to click **Ready** before the game can progress to the next scene (before the Host can start the game). After players click Ready, you send a `ServerRPC` `OnClientIsReadyServerRpc` (inside the `PlayerIsReady` function). When it arrives server-side, it marks the client state as a ready based on its `ClientId` (line 218. You keep track if a client is ready in the `m_ClientsInLobby` `Dictionary`.
+
+```csharp
+/// <summary>
+///     OnClientIsReadyServerRpc
+///     Sent to the server when the player clicks the ready button
+/// </summary>
+/// <param name="clientid">clientId that is ready</param>
+[ServerRpc(RequireOwnership = false)]
+private void OnClientIsReadyServerRpc(ulong clientid)
+{
+    if (m_ClientsInLobby.ContainsKey(clientid))
+    {
+           m_ClientsInLobby[clientid] = true;
+        UpdateAndCheckPlayersInLobby();
+        GenerateUserStatsForLobby();
+    }
+}
+```
+
+<!-- ```csharp reference
+https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/LobbyControl.cs#L208-L222
+``` -->
+
+At the same time that you sync the Clients to update their UI, send a ClientRpc to update the other players' ready state. The update is handled by the ClientRpc called `SendClientReadyStatusUpdatesClientRpc` in `UpdateAndCheckPlayersInLobby`.
```

Comment:
Replace this with:

"At the same time, in order to sync up with the rest of the clients and update their UI, we send a ClientRpc. The update is handled by the ClientRpc called SendClientReadyStatusUpdatesClientRpc in UpdateAndCheckPlayersInLobby."